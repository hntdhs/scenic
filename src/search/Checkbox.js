import React, { useState } from "react";

function Checkbox(props) {
    const [checked, setChecked] = useState(false);

    const handleChange = () => {
        setChecked(!checked);
        props.onChange({'checked': !checked, 'value': props.name})
    };
    // wasn't handling the useState or handleChange here before and it didn't work; always best to handle it more directly like this

    return (
        <div>
            <label>
                <input
                    type="checkbox"
                    checked={checked}
                    onChange={handleChange}
                    name={props.name}
                    value={props.name}
                    class="form-check-input"
                />
            </label>
        </div>
    );
};

export default Checkbox;