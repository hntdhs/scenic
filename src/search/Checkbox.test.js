import React from "react";
import { render } from "@testing-library/react";
import Checkbox from "./Checkbox.js"

it("matches snapshot", function () {
    const { asFragment } = render(<Checkbox />);
    expect(asFragment()).toMatchSnapshot();
  });