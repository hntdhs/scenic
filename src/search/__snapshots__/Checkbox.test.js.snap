// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <label>
      <input
        type="checkbox"
        value=""
      />
    </label>
  </div>
</DocumentFragment>
`;
